docs/topics/native/native-definition-file.md

```diff
@@ -0,0 +1,280 @@
+| [`foreignExceptionMode`](#wrap-exceptions)                                          | Wraps exceptions from Objective-C code into Kotlin exceptions with the `ForeignException` type.                                    |
+| `pluginName`                                                                        |                                                                                                                                    |
+| `objcClassesIncludingCategories`                                                    |                                                                                                                                    |
+| [`userSetupHint`](#linker-errors)                                                   | Adds a custom message, for example, to help users resolve linker errors.                                                           |
+
+In addition to the list of properties, you can include [custom declarations](#add-custom-declarations) in your definition
+file.
+
+### Headers
+
+When the library is imported, all of the headers are also included in the program. So all header dependencies are included
+in generated stubs as well.
+
+This behavior can be very inconvenient for some libraries. Use the `headers` property to specify separate headers that
+should be imported:
+
+```none
+headers = curl/curl.h
+```
+
+### Filter headers by globs
+
+You can filter headers by globs using filter properties from the `.def` file. To include declarations from headers,
+use the `headerFilter` property. If a header matches any of the globs, its declarations are included in the bindings.
+
+The globs are applied to the header paths relative to the appropriate include path elements,
+for example, `time.h` or `curl/curl.h`. So if the library is usually included with `#include <SomeLibrary/Header.h>`,
+it would probably be correct to filter headers with the following filter:
+
+```none
+headerFilter = SomeLibrary/**
+```
+
+If `headerFilter` is not provided, all the headers are included. However, we encourage you to use `headerFilter`
+and specify the glob as precisely as possible. In this case, the generated library contains only the necessary
+declarations. It can help avoid various issues when upgrading Kotlin or tools in your development environment.
+
+### Exclude headers
+
+To exclude specific headers, use the `excludeFilter` property. It can be helpful to remove redundant or problematic
+headers and optimize compilation, as declarations from the specified headers are not included into the bindings:
+
+```none
+excludeFilter = SomeLibrary/time.h
+```
+
+> If the same header is both included with `headerFilter`, and excluded with `excludeFilter`, the specified header will
+> not be included into the bindings.
+>
+{type="note"}
+
+### Filter headers by module maps
```

Comment:
I guess this section should focus on the `modules` property, not the obscure `excludeDependentModules`.